### PR TITLE
crio: Fixup docker SELinux permissions

### PR DESCRIPTION
**Pull request by mrunalp**
@rhatdan @sdodson @ashcrow  PTAL
Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by ashcrow** [comment]
Can you verify this shouldn't be ```/var/lib/containers/storage/overlay2```?

**Comment by mrunalp** [comment]
@ashcrow This is fine as it is meant for the docker storage directory which is at `/var/lib/containers/overlay2`

File: roles/container_runtime/tasks/package_crio.yml [hunk 1]
```diff
@@ -69,6 +69,11 @@
     dest: /etc/sysconfig/crio-network
     src: crio-network.j2
 
+- name: Fixup SELinux permissions for docker
+  shell: |
+          semanage fcontext -a -e /var/lib/docker/overlay2 /var/lib/containers/overlay2
+          restorecon -R -v /var/lib/containers/overlay2
```
Comment (rhatdan) [inline roles/container_runtime/tasks/package_crio.yml]: NIT: The -v is for verbose output.  Not sure it is needed, but it will not hurt anything.

**Comment by rhatdan** [comment]
Correct this is for /var/lib/containers/overlay2 which is docker storage not containers storage.

**Comment by sdodson** [comment]
How do we stop making the installer responsible for this sort of stuff? Shouldn't this come out of the box in a working manner?

**Comment by ashcrow** [comment]
> How do we stop making the installer responsible for this sort of stuff? Shouldn't this come out of the box in a working manner?

The problem is it's "easy" to do a quick fix in the installer and forget then change the project code, rebuild, push, and test. It **should** be correct out of the box and, in a more perfect world, would have been seen in testing and/or CI. I *think* the best way to keep this from happening is to have more/better test cases with the expected released ``container-selinux`` version at the project layer and the openshift CI layer.

I echo your sentiment ... the installer shouldn't be the gatekeeper of selinux work for openshift in general.

**Comment by rhatdan** [comment]
The standard location for docker overlay2 is out of the box supported.  You are modifying the standard to have storage in /var/lib/containers/overlay2 rather then /var/lib/docker/overlay2.  The rule you are adding says to setup labeling of /var/lib/containers/overlay2 as if it was /var/lib/docker/overlay2.  Not sure how your ansible script customizing the location, could be expected to have proper labeling, unless your script also sets up the labeling.

**Comment by ashcrow** [comment]
@rhatdan ah, that makes sense.

**Comment by sdodson** [comment]
Ok, I need to figure out the history here and why we're currently straying from the norm and what we can do to get back to just running with the defaults.

**Comment by ashcrow** [comment]
@sdodson did @mwoodson answer the background questions from your POV?

Failures looks unrelated/flake:

```
last_checked_host: localhost, last_checked_var: openshift_image_tag;openshift_image_tag must be in the format
               v#.#.#[-optional.#]. Examples: v1.2.3, v3.5.1-alpha.1
               You specified openshift_image_tag=6160a5e
```

**Comment by phemmer** [comment]
I was just about to add a PR to delete this task when I noticed that it was just added. Please see #7715

**Pull request merged**